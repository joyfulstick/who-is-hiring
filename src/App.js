import React from 'react'

const App = () => <p>Who is hiring</p>

export default App
