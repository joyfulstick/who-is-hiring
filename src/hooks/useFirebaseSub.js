import firebase from 'firebase/app'
import 'firebase/database'
import { useEffect, useState } from 'react'

firebase.initializeApp({
  databaseURL: 'https://hacker-news.firebaseio.com',
})

export function useFirebaseSub(path) {
  const dbRef = firebase
    .database()
    .ref()
    .child(path)
  const [state, setState] = useState(null)
  useEffect(() => {
    dbRef.on('value', s => setState(s.val()))
    return () => {
      dbRef.off()
    }
  }, [])

  return state
}
