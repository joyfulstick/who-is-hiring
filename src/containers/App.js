import 'firebase/database'
import React, { useEffect, useState } from 'react'
import Filters from '../components/Filters'
import useFirebaseSub from '../hooks/useFirebaseSub'
import HeaderStyle from '../styles/HeaderStyle'
import MainStyle from '../styles/MainStyle'
import List from './List'

const App = () => {
  const [stories, setStories] = useState(null)

  const jobstories = useFirebaseSub('/v0/jobstories')
  useEffect(() => {
    setStories(jobstories)
  }, [jobstories])

  const [filters, setFilters] = useState({
    position: 'all',
    salary: 5,
    isRemote: false,
  })

  const handleChange = ({ target: { type, name, value, checked } }) => {
    const v = type === 'checkbox' ? checked : value
    setFilters({ ...filters, [name]: v })
  }

  return (
    <>
      <HeaderStyle>
        <Filters handleChange={handleChange} filters={filters} />
      </HeaderStyle>
      <MainStyle>
        {!stories ? (
          <p>LOADING LIST...</p>
        ) : (
          stories.map(s => <List key={s} storie={s} filters={filters} />)
        )}
      </MainStyle>
    </>
  )
}

export default App
