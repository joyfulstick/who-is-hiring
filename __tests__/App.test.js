//TODO: rerender <App /> with an updated state

import React from 'react'
// import ReactDOM from 'react-dom'
import App from '../src/containers/App'
import { shallow } from 'enzyme'
import '@babel/polyfill'

// jest.mock('../src/hooks/useFirebaseSub')
// const foo = require('../src/hooks/useFirebaseSub')
// foo.mockImplementation(() => ['42'])

// it('renders without crashing', () => {
//   const div = document.createElement('div')
//   ReactDOM.render(<App />, div)
//   ReactDOM.unmountComponentAtNode(div)
// })
// jest.mock('../src/hooks/useFirebaseSub', () =>
//   jest.fn(() => ['1', '2']),
// )
// jest.mock('../src/hooks/useFirebaseSub', () => ({
//   useFirebaseSub: jest.fn(),
// }))

// import { useFirebaseSub } from '../src/hooks/useFirebaseSub'
// useFirebaseSub.mockImplementation(() => null)
// await useFirebaseSub.mockImplementation(() => ['1', '2'])

jest.mock('../src/hooks/useFirebaseSub')

import { useFirebaseSub } from '../src/hooks/useFirebaseSub'

it('renders corretly', async () => {
  const r = new Response(['1', '2'])
  useFirebaseSub.mockReturnValue(Promise.resolve(r))
  const wrapper = shallow(<App />)
  // wrapper.update()

  expect(wrapper).toMatchSnapshot()
})
