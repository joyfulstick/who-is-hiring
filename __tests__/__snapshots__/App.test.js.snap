// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corretly 1`] = `
<Fragment>
  <styled.header>
    <Filters
      filters={
        Object {
          "isRemote": false,
          "position": "all",
          "salary": 5,
        }
      }
      handleChange={[Function]}
    />
  </styled.header>
  <styled.main>
    <p>
      LOADING LIST...
    </p>
  </styled.main>
</Fragment>
`;
